applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -287,39 +308,43 @@ static int device_pres_delay_find(uint8_t index, uint32_t *pres_dly_us)
  * @retval	0	Operation successful.
  * @retval	-EINVAL	There is no match.
  */
-static int device_index_get(const struct bt_conn *conn, uint8_t *index)
+static int device_index_get(const struct bt_conn *conn, struct stream_index *stream_idx)
 {
 	if (conn == NULL) {
 		LOG_ERR("No connection provided");
 		return -EINVAL;
 	}
 
-	for (int i = 0; i < ARRAY_SIZE(unicast_servers); i++) {
-		if (unicast_servers[i].device_conn == conn) {
-			*index = i;
-			return 0;
+	for (int i = 0; i < CONFIG_BT_ISO_MAX_CIG; i++) {
+		for (int j = 0; j < ARRAY_SIZE(unicast_servers[i][0]); j++) {
+			if (unicast_servers[i][0][j].device_conn == conn) {
+				stream_idx->lvl1_idx = i;
+				stream_idx->lvl2_idx = 0;
+				stream_idx->lvl3_idx = j;
+				return 0;
+			}
 		}
 	}
 
-	LOG_WRN("Connection not found");
```

Comment:
We should still have a warning here in case, for some reason, the if conditional does not trigger.

Reply:
We use this function in the `device_index_vacant_get()`, so every time we need a vacant spot, meaning a new connection, we will get this warning print. I instead added warning/error prints to all other places that use this function